patches/api/0453-Brigadier-based-command-API.patch

```diff
@@ -0,0 +1,1090 @@
++
++    /**
++     * Executes the command with the given {@link CommandSourceStack} and arguments.
++     *
++     * @param commandSourceStack The commandSourceStack of the command.
++     * @param args The arguments of the command.
++     * @return The result of the command execution.
++     */
++    @ApiStatus.OverrideOnly
++    int execute(@NotNull CommandSourceStack commandSourceStack, @NotNull String[] args);
++
++    /**
++     * Suggests possible completions for the given command {@link CommandSourceStack} and arguments.
++     *
++     * @param commandSourceStack The commandSourceStack of the command.
++     * @param args The arguments of the command.
++     * @return A collection of suggestions
++     */
++    @ApiStatus.OverrideOnly
++    default @NotNull Collection<String> suggest(final @NotNull CommandSourceStack commandSourceStack, final @NotNull String[] args) {
++        return Collections.emptyList();
++    }
++}
+diff --git a/src/main/java/io/papermc/paper/command/brigadier/CommandSourceStack.java b/src/main/java/io/papermc/paper/command/brigadier/CommandSourceStack.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..fee4a3a1ab20aa611b5ad13bc30127833a6a0ed9
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/command/brigadier/CommandSourceStack.java
+@@ -0,0 +1,54 @@
++package io.papermc.paper.command.brigadier;
++
++import com.mojang.brigadier.context.CommandContext;
++import com.mojang.brigadier.exceptions.CommandSyntaxException;
++import io.papermc.paper.command.brigadier.argument.ArgumentResolver;
++import org.bukkit.Location;
++import org.bukkit.World;
++import org.bukkit.command.CommandSender;
++import org.bukkit.entity.Entity;
++import org.jetbrains.annotations.ApiStatus;
++import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Nullable;
++
++/**
++ * Represents a vanilla command source stack which is used
++ * for command execution.
++ */
++@ApiStatus.NonExtendable
++@ApiStatus.Experimental
++public interface CommandSourceStack {
++
++    /**
++     * Gets the location that this command is being executed at.
++     * @return location
++     */
++    @NotNull Location getLocation();
++
++    /**
++     * Gets the source of from where this command was executed from
++     * @return source type
++     */
++    @NotNull CommandSender getSender();
++
++    /**
++     * Gets the entity that triggered the execution of this command.
++     * May not always be the command source.
++     */
++    @Nullable Entity getExecutor();
++
++    /**
++     * Resolves an argument in this command source stack.
++     *
++     * @param context command context
++     * @param name name of argument
++     * @return resolved argument
++     * @param <V> resolved argument type
++     */
++    @SuppressWarnings("unchecked")
++    default <V> @NotNull V getResolvedArgument(final @NotNull CommandContext<CommandSourceStack> context, final @NotNull String name) throws CommandSyntaxException {
```

Comment:
`argument`, `resolvedArgument`, just `get`, or something else that's shorter than that full thing